formats/dsd/http_test.go

```diff
@@ -23,14 +23,23 @@ func TestMimeTypes(t *testing.T) {
 	}
 
 	// Test assumptions.
-	for mimeType, mimeTypeCleaned := range map[string]string{
-		"application/xml, image/webp":       "xml",
-		"application/xml;q=0.9, image/webp": "xml",
-		"*":                                 "*",
-		"*/*":                               "*",
-		"text/yAMl":                         "yaml",
+	for accept, format := range map[string]uint8{
+		"application/json, image/webp":       JSON,
+		"image/webp, application/json":       JSON,
+		"application/json;q=0.9, image/webp": JSON,
+		"*":                                  DefaultSerializationFormat,
+		"*/*":                                DefaultSerializationFormat,
+		"text/yAMl":                          YAML,
+		" * , yaml ":                         YAML,
+		"yaml;charset ,*":                    YAML,
+		"xml,*":                              DefaultSerializationFormat,
+		"text/xml, text/other":               AUTO,
+		"text/*":                             DefaultSerializationFormat,
+		"yaml ;charset":                      AUTO, // Invalid mimetype format.
+		"":                                   DefaultSerializationFormat,
+		"x":                                  AUTO,
 	} {
```

Comment:
That list seems a bit weird, especially the `chartset=` and `q=` parts. Can we make that a bit more obvious on why the strings look like this?